libraries/Microsoft.Bot.Builder/Streaming/StreamingRequestHandler.cs

```diff
@@ -381,11 +383,24 @@ private async Task ReconnectAsync(IDictionary<string, string> requestHeaders = n
                 }
             }
 
+            // Set the authentication header if it wasn't passed in.
+            if (requestHeaders != null && !requestHeaders.ContainsKey("authentication") && !requestHeaders.ContainsKey("Authentication"))
+            {
+                try
+                {
+                    clientWebSocket.Options.SetRequestHeader("authentication", await _appCredentials.GetTokenAsync().ConfigureAwait(false));
+                }
+                catch (Exception e)
+                {
+                    _logger.LogError(e.Message);
+                }                
+            }            
+
             // The ServiceUrl of a streaming connection follows the pattern "urn:[ChannelName]:[Protocol]:[Host]".
             var uri = ServiceUrl.Split(':');
             var protocol = uri[uri.Length - 2];
```

Comment:
These 2 should be validated. We cannot guarantee the shape of ServiceUrl, so we need to validate here and throw if it doesn;t meet what we expect

Reply:
Added validation and a catch all exception message, let me know if it should be broken out into three checks and use a canned exception.